external_api/lib/dictionary_utils.py

```diff
@@ -25,45 +47,62 @@ def dictionary_update_languages(dictionary_name, new_language_list):
         raise APIHandlerException('No new languages provided. Nothing changed.')
 
     # fetch all words
-    word_list = get_dictionary_unique_words(dictionary_name)
-    if not word_list:
-        raise APIHandlerException('This dictionary does not have any records. Please verify the dictionary name')
+    values = get_entity_unique_values(entity_name=entity_name)
+    if not values:
+        raise APIHandlerException('This entity does not have any records. Please verify the entity name')
 
     records_to_create = []
     for language_script in languages_added:
         # create records for all words
-        for word in word_list:
+        for value in values:
             records_to_create.append({
-                'word': word,
+                'word': value,
                 'language_script': language_script,
                 'variants': []
             })
 
     datastore_obj = DataStore()
-    datastore_obj.add_data_elastic_search(dictionary_name, records_to_create)
+    datastore_obj.add_entity_data(entity_name, records_to_create)
 
     return True
 
 
-def get_dictionary_unique_words(
-    dictionary_name, empty_variants_only=False, word_search_term=None, variant_search_term=None
+def get_entity_unique_values(
+    entity_name, empty_variants_only=False, value_search_term=None, variant_search_term=None
 ):
     """
+    Get a list of unique values belonging to this entity
+    Args:
+        entity_name (str): Name of the entity for which unique values are to be fetched
+        empty_variants_only (bool): Flag to search for values with empty variants only
+        value_search_term (str): Search term to filter values from this entity data
+        variant_search_term (str): Search term to filter out variants from the entity data
```

Comment:
Minor,
lines 77 - 79
optional indicator in doc, and what happens with default values